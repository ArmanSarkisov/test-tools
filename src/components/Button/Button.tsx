import styles from './Button.module.scss'

export const Button = () => {
    return <button className={styles.button} type="button">My custom button</button>
}