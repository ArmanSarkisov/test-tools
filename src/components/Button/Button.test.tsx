import { describe, it } from 'vitest'
import { render, screen } from '@testing-library/react'

import { Button } from './Button'

describe('suite name', () => {
    it('should render component', () => {
        render(<Button />);
        const headline = screen.getByText(/My custom button/i);
        expect(headline).toBeInTheDocument();
        expect(headline).toMatchSnapshot();
    })
})