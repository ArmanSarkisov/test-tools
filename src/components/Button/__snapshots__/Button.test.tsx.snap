// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`suite name > should render component 1`] = `
<button
  class="_button_f8f296"
  type="button"
>
  My custom button
</button>
`;
